x/staking/keeper/msg_server.go

```diff
@@ -309,17 +312,14 @@ func (k msgServer) BeginRedelegate(goCtx context.Context, msg *types.MsgBeginRed
 		}
 	}
 
-	// If this delegation is from a liquid staking provider, the delegation on the new validator
+	// If this delegation from a liquid staker, the delegation on the new validator
 	// cannot exceed that validator's self-bond cap
 	// The liquid shares from the source validator should get moved to the destination validator
-	if k.AccountIsLiquidStakingProvider(delegatorAddress) {
-		if err := k.SafelyIncreaseValidatorTotalLiquidShares(ctx, &dstValidator, dstShares); err != nil {
-			return nil, err
-		}
-
-		if err := k.DecreaseValidatorTotalLiquidShares(ctx, &srcValidator, srcShares); err != nil {
+	if k.DelegatorIsLiquidStaker(delegatorAddress) {
+		if err := k.SafelyIncreaseValidatorLiquidShares(ctx, &dstValidator, dstShares); err != nil {
 			return nil, err
 		}
+		k.DecreaseValidatorLiquidShares(ctx, &srcValidator, srcShares)
```

Comment:
Should we check if func return any err here?

Reply:
Good catch